### PR TITLE
Closes #486: CopyFile logs incorrect targetFilePath

### LINKED ISSUE #486: BUG: CopyFile logs incorrect targetFilePath
The log message written by `FileCopier.CopyFileCore` reports the source file path as the target file path:
```c#
context.Log.Verbose("Copying file {0} to {1}", absoluteFilePath.GetFilename(), absoluteFilePath);
```
So the log message makes it look like the file is just getting copied to itself (although the copy actually works correctly).

### PR DESCRIPTION
**Pull request by jrnail23**

### CODE REVIEW

**Comment by devlead** [comment]
Looks sold @jrnail23 :+1: , could you get the branch up to date and I'll merge.

**Comment by jrnail23** [comment]
@devlead, should be ready to go now, pending CI build

**Comment by patriksvensson** [comment]
Restarted Linux build that had timed out when restoring NuGet packages.

**Comment by jrnail23** [comment]
@patriksvensson, did it happen again?

**Comment by patriksvensson** [comment]
@jrnail23 Hmm. Yes :cry: I'll restart it again.

I should really try to get the packages properly cached on Travis since this fail in more than 50% of the builds.

File: src/Cake.Common.Tests/Unit/IO/FileAliasesTests.cs [hunk 1]
```diff
@@ -200,6 +202,22 @@ public void Should_Copy_File()
...
             }
+
+            [Fact]
+            public void Should_Log_Verbose_Message_With_Correct_Target()
+            {
+                // Given
+                var fixture = new FileCopyFixture();
+
+                // When
+                FileAliases.CopyFile(fixture.Context, "./file1.txt", "./target/file1.txt");
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/IO/FileAliasesTests.cs]: Sorry for nitpicking, but a better test would be to test for `./file1.txt` and `./target/file2.txt`.
If I understand correctly the bug will output the filename of the target file first, which would give the same results.

Comment (jrnail23) [inline src/Cake.Common.Tests/Unit/IO/FileAliasesTests.cs]: I agree, but the fixture isn't very straightforward in terms of changing that part.  I did, however, start from a failing test.
I can still try to get that to work if you like.

Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/IO/FileAliasesTests.cs]: Ah, I see. I think it's OK then :smile:

**Comment by patriksvensson** [comment]
This looks good to me :+1: 
Ok to merge?

**Comment by patriksvensson** [comment]
@jrnail23 Looks like this needs to be rebased against develop.

**Comment by jrnail23** [comment]
@patriksvensson, rebased again, ready to go (pending CI build)

**Comment by devlead** [comment]
Merged @jrnail23, thanks for contributing :+1:

**Comment by jrnail23** [comment]
:+1:

**Pull request merged**